main.go

```diff
@@ -4,27 +4,28 @@ import (
 	"encoding/json"
 
 	"github.com/mount-joy/thelist-lambda/handlers"
+	"github.com/mount-joy/thelist-lambda/handlers/iface"
 
 	"github.com/aws/aws-lambda-go/events"
 	"github.com/aws/aws-lambda-go/lambda"
 )
 
 type handler struct {
-	router handlers.Router
+	router iface.Router
 }
 
-func (h *handler) doRequest(request events.APIGatewayV2HTTPRequest) (events.APIGatewayProxyResponse, error) {
+func (h *handler) doRequest(request events.APIGatewayV2HTTPRequest) (events.APIGatewayV2HTTPResponse, error) {
 	result, statusCode := h.router.Route(request)
 
 	res, err := json.Marshal(result)
 	if err != nil {
-		return events.APIGatewayProxyResponse{
-			Body:       err.Error(),
+		return events.APIGatewayV2HTTPResponse{
+			Body:       "{\"error\": \"" + err.Error() + "\"}",
```

Comment:
Could use back ticks instead of double quotes then you don't need to escape all of the double quotes
```
`{"error": "` + err.Error() + `"}`
```
Possible that sprintf would be more clear, although slightly less performant:
`fmt.Sprintf("{\"error\": %q}", error.Error())`
The `%q` wraps it in double quotes